bee-storage/bee-storage/src/health.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines a type to represent different health states in which the storage backend can be.
+
+use bee_common::packable::{Packable, Read, Write};
+
+/// Errors related to storage health statuses.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// I/O error happened.
+    #[error("I/O error happened: {0:?}")]
+    Io(#[from] std::io::Error),
+    /// Unknown storage health variant.
+    #[error("Unknown storage health variant: {0}")]
+    UnknownHealthVariant(u8),
```

Comment:
I think `UnknownHealth` is a better variant name, otherwise we would have to talk about the `UnknownHealthVariant`-variant, which feels odd.